infra/feast-operator/internal/controller/services/repo_config.go

```diff
@@ -44,71 +45,154 @@ func (feast *FeastServices) getServiceFeatureStoreYaml(feastType FeastServiceTyp
+			if err != nil {
+				return err
+			}
+		}
+	}
+	repoConfig.Registry = RegistryConfig{
+		RegistryType:       RegistryConfigType(persistenceType),
+		Path:               filePath,
+		S3AdditionalKwargs: s3AdditionalKwargs,
+		DBParameters:       parametersMap,
+	}
+	repoConfig.OfflineStore = OfflineStoreConfig{}
+	repoConfig.OnlineStore = OnlineStoreConfig{}
+
+	return nil
+}
+
+func setRepoConfigOnline(services *feastdevv1alpha1.FeatureStoreServices, secretExtractionFunc func(secretRef string, secretKeyName string) (map[string]interface{}, error), repoConfig *RepoConfig) error {
+	var filePath = DefaultOnlineStoreEphemeralPath
+	var parametersMap map[string]interface{}
+	persistenceType := string(OnlineSqliteConfigType)
+
+	if services.OnlineStore.Persistence != nil && services.OnlineStore.Persistence.FilePersistence != nil {
+		filePersistence := services.OnlineStore.Persistence.FilePersistence
+		filePath = getActualPath(filePersistence.Path, filePersistence.PvcConfig)
+	} else if services.OnlineStore.Persistence != nil &&
+		services.OnlineStore.Persistence.DBPersistence != nil {
+		if len(services.OnlineStore.Persistence.DBPersistence.Type) > 0 {
+			var err error
+			filePath = ""
+			persistenceType = services.OnlineStore.Persistence.DBPersistence.Type
+			secretKeyName := services.OnlineStore.Persistence.DBPersistence.SecretKeyName
+			if len(secretKeyName) == 0 {
+				secretKeyName = persistenceType
+			}
+			parametersMap, err = secretExtractionFunc(services.OnlineStore.Persistence.DBPersistence.SecretRef.Name, secretKeyName)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	repoConfig.OnlineStore = OnlineStoreConfig{
+		Type:         OnlineConfigType(persistenceType),
+		Path:         filePath,
+		DBParameters: parametersMap,
+	}
+
+	return nil
+}
+
+func setRepoConfigOffline(services *feastdevv1alpha1.FeatureStoreServices, secretExtractionFunc func(secretRef string, secretKeyName string) (map[string]interface{}, error), repoConfig *RepoConfig) error {
+	var parametersMap map[string]interface{}
+	persistenceType := string(OfflineFilePersistenceDaskConfigType)
+
+	if services.OfflineStore.Persistence != nil &&
+		services.OfflineStore.Persistence.FilePersistence != nil &&
+		len(services.OfflineStore.Persistence.FilePersistence.Type) > 0 {
+		persistenceType = services.OfflineStore.Persistence.FilePersistence.Type
+	} else if services.OfflineStore.Persistence != nil &&
+		services.OfflineStore.Persistence.DBPersistence != nil {
+		if len(services.OfflineStore.Persistence.DBPersistence.Type) > 0 {
+			var err error
+			persistenceType = services.OfflineStore.Persistence.DBPersistence.Type
+			secretKeyName := services.OfflineStore.Persistence.DBPersistence.SecretKeyName
+			if len(secretKeyName) == 0 {
+				secretKeyName = persistenceType
+			}
+			parametersMap, err = secretExtractionFunc(services.OfflineStore.Persistence.DBPersistence.SecretRef.Name, secretKeyName)
+			if err != nil {
+				return err
+			}
+		}
+	}
```

Comment:
for these setRepoConfig functions, maybe we can clean it up a bit by consolidating?
```suggestion
        offlinePersistence := services.OfflineStore.Persistence
	if offlinePersistence != nil &&
		offlinePersistence.FilePersistence != nil &&
		len(offlinePersistence.FilePersistence.Type) > 0 {
		persistenceType = offlinePersistence.FilePersistence.Type
	} else if offlinePersistence != nil &&
		offlinePersistence.DBPersistence != nil {
		if len(offlinePersistence.DBPersistence.Type) > 0 {
			var err error
			persistenceType = offlinePersistence.DBPersistence.Type
			secretKeyName := offlinePersistence.DBPersistence.SecretKeyName
			if len(secretKeyName) == 0 {
				secretKeyName = persistenceType
			}
			parametersMap, err = secretExtractionFunc(offlinePersistence.DBPersistence.SecretRef.Name, secretKeyName)
			if err != nil {
				return err
			}
		}
	}
```

Reply:
Done